BookingsApi/BookingsApi/Controllers/HearingsController.cs

```diff
@@ -848,6 +793,85 @@ public async Task<IActionResult> GetBookingStatusById(Guid hearingId)
         }
 
 
+        private string BuildCursorPageUrl(
+            string cursor,
+            int limit,
+            List<int> caseTypes,
+            string caseNumber = "",
+            List<int> hearingVenueIds = null,
+            string lastName = "")
+        {
+            const string hearingsListsEndpointBaseUrl = "hearings/";
+            const string bookingsEndpointUrl = "types";
+            const string resourceUrl = hearingsListsEndpointBaseUrl + bookingsEndpointUrl;
+
+            var types = string.Empty;
+            if (caseTypes.Any())
+            {
+                types = string.Join("&types=", caseTypes);
+            }
+
+            var pageUrl = $"{resourceUrl}?types={types}&cursor={cursor}&limit={limit}";
+
+            // Executes when Admin_Search feature toggle is ON and search action is performed
+            if (_featureToggles.AdminSearchToggle())
```

Comment:
you could reduce the cognitive complexity the code by reducing the nested statements. one common approach is to return early.

```suggestion
            if (!_featureToggles.AdminSearchToggle()) return;
```

Reply:
this method is not part of this ticket. Happy to raise an improvement ticket.

Reply:
I don't think we need a separate ticket for improvements around code you are working around or on. if you still don't wish to make this change then please add a ticket so we can get back to it

Reply:
ok ticket raised https://tools.hmcts.net/jira/browse/VIH-9671